src/amo/components/Link/index.js

```diff
@@ -95,7 +95,10 @@ export class LinkBase extends React.Component {
     };
 
     if (typeof href === 'string') {
-      const linkHref = urlPrefix ? joinUrl.pathname(urlPrefix, href) : href;
+      const linkHref =
+        urlPrefix && !href.startsWith(urlPrefix)
```

Comment:
`prependClientApp` etc. are only passed as `false` for an external link, but if a link is interpreted as internal (e.g., `https://addons-dev.allizom.org/en-US/firefox/addon/soundfixer/` when on the dev site), then `prependClientApp` etc. are *not* passed as `false`, so this is to protect against that case.

Reply:
Hmm, ok. You could just always pass `prependClientApp: false` and `prependLang: false`, though? I guess it's fine as is.

Reply:
No, I can't always pass that, because the URL might be a "relative" URL, e.g., `/addon/slug`, and in that case I *do* want the `clientApp` and `lang` prepended. When we have an internal URL we don't really know whether it's going to need prepending or not.